mne/io/write.py

```diff
@@ -129,7 +129,7 @@ def write_julian(fid, kind, data):
 
 def write_string(fid, kind, data):
     """Write a string tag."""
-    str_data = data.encode('latin1')
+    str_data = str(data).encode('latin1')
```

Comment:
Note about this change:

When writing an `src`, the field `"mri_volume_name"` could be a `Path` which can not be encoded to `"latin1"`.
IMO, it is safer to cast to string before writing than to chase every case where something else than a string might be provided to `write_string`.

Reply:
agreed that casting to `str` before write operations is the correct fix. I think I've hit this problem before (and obviously didn't fix it everywhere 😉).